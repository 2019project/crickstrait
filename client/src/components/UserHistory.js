//piyush
import React, { Component } from "react";
import "../styles/UserHistory.css";
import UserNavigation from "./Common/UserNavigation";
import { connect } from "react-redux";
import { userHistory } from "../actions/Users";
import jwt_decode from "jwt-decode";
// for user to see the history of hid stock buy and sell

export class UserHistory extends Component {
  componentDidMount() {
    if (localStorage.getItem("token")) {
      var decode = jwt_decode(localStorage.getItem("token"));
      this.props.userHistory(decode.email);
    }
  }
  render() {
    return (
      <div>
        <div id="userHistoryContainer">
          <UserNavigation selected="history" />
          {/*side bar for user detail*/}
          <div id="userhistory">
            {this.props.userhistory.length != 0 ? (
              <table id="userhistory_table">
                <thead id="userhistory_tableHeader">
                  <th>Stocks</th>
                  <th>Price</th>
                  <th>Buy/Sell</th>
                  <th>Quantity</th>
                  <th>Date</th>
                </thead>
                {this.props.userhistory.map(hist => (
                  <>
                    {hist.company.map((stocks, index) => (
                      <tr>
                        <td id={"tickerName" + index}>{stocks.ticker_name}</td>
                        <td id={"tickerPrice" + index}>
                          {stocks.current_price}
                        </td>
                        {stocks.buy == true ? <td>Bought</td> : <td>Sold</td>}
                        <td id={"tickerQty" + index}>
                          {stocks.buying_quantity}
                        </td>
                        <td id={"tickerDate" + index}>
                          {new Date(stocks.buy_date).toLocaleDateString(
                            "en-In",
                            {
                              month: "short",
                              day: "2-digit",
                              year: "numeric"
                            }
                          )}
                        </td>
                      </tr>
                    ))}
                  </>
                ))}
              </table>
            ) : (
              <h1>No Stocks History</h1>
            )}
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  userhistory: state.usersReducer.userhistory,
  isLoading: state.LoadingReducer.isLoading
});
export default connect(mapStateToProps, {
  userHistory
})(UserHistory);
