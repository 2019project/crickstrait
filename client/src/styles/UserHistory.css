@import url(https://fonts.googleapis.com/css?family=Karla);
* {
  font-family: Karla;
}
#userHistoryContainer {
  width: 100%;
  display: grid;
  grid-template-rows: 150px auto;
  font-family: Karla;
  text-align: center;
}
#userhistory {
  background: url("../components/Common/stocks_wallpaper.jpg") no-repeat;
  background-size: cover;
  margin-top: 60px;
  height: 92vh;
}

#userhistory h1 {
  text-align: center;
  font-weight: 700;
  font-family: Karla;
  font-size: 40px;
  margin-top: 200px;
  color: #2c3e50;
  text-shadow: 3px 3px 6px #707070;
}
#userhistory_table {
  width: 90%;
  font-size: 12px;
  height: auto;
  background-color: #fff;
  border-collapse: collapse;
  box-shadow: 4px 4px 6px #707070;
  border-radius: 4px;
  margin: 30px auto;
}
#userhistory_tableHeader {
  background-color: #2c3e50;
  color: #fff;
  border-radius: 8px;
  width: 80%;
}

#userhistory_table th {
  padding: 10px;
  color: #fff;
}

#userhistory_table td {
  padding: 10px;
  text-align: center;
  color: #2c3e50;
}

@media screen and (min-width: 430px) {
  #userHistoryContainer {
    width: 100%;
    grid-template-columns: 250px 1fr;
    font-family: Karla;
    margin: auto;
  }

  #userhistory {
    margin-top: 0px;
  }

  #userhistory_table {
    width: 55%;
    font-size: 16px;
    height: auto;
    background-color: #fff;
    border-collapse: collapse;
    border-radius: 4px;
    margin: 130px auto;
  }
}
