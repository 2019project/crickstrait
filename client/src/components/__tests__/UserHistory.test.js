/* useer history component test case -piyush*/
import React from "react";
import { shallow, mount } from "enzyme";
import { UserHistory } from "../UserHistory";

const userHistory = jest.fn();
const userhistory = [
  {
    company: [
      {
        buy: true,
        buy_date: "2019-11-22T07:27:35.875Z",
        buying_quantity: "7",
        current_price: 201.55,
        sell: false,
        ticker_name: "AAPL"
      },
      {
        buy: true,
        buy_date: "2019-11-22T10:22:26.502Z",
        buying_quantity: "6",
        current_price: 135.68,
        sell: false,
        ticker_name: "MSFT"
      }
    ]
  }
];

const wrapper = shallow(
  <UserHistory userhistory={userhistory} userHistory={userHistory} />
);

describe("Testing User History Component", () => {
  it("should mount the component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should have a tickername in each row ", () => {
    expect(wrapper.find("#tickerName0").props().children).toBe(
      userhistory[0].company[0].ticker_name
    );
  });

  it("should have a tickerPrice in each row ", () => {
    expect(wrapper.find("#tickerPrice0").props().children).toBe(
      userhistory[0].company[0].current_price
    );
  });

  it("should have a tickerOty in each row ", () => {
    expect(wrapper.find("#tickerQty0").props().children).toBe(
      userhistory[0].company[0].buying_quantity
    );
  });

  it("should have a tickerDate in each row ", () => {
    expect(wrapper.find("#tickerDate0").props().children).toBe(
      new Date(userhistory[0].company[0].buy_date).toLocaleDateString("en-In", {
        month: "short",
        day: "2-digit",
        year: "numeric"
      })
    );
  });
});
