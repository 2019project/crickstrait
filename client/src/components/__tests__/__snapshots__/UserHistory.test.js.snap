// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing User History Component should mount the component 1`] = `
<div>
  <div
    id="userHistoryContainer"
  >
    <Connect(UserNavigation)
      selected="history"
    />
    <div
      id="userhistory"
    >
      <table
        id="userhistory_table"
      >
        <thead
          id="userhistory_tableHeader"
        >
          <th>
            Stocks
          </th>
          <th>
            Price
          </th>
          <th>
            Buy/Sell
          </th>
          <th>
            Quantity
          </th>
          <th>
            Date
          </th>
        </thead>
        <tr>
          <td
            id="tickerName0"
          >
            AAPL
          </td>
          <td
            id="tickerPrice0"
          >
            201.55
          </td>
          <td>
            Bought
          </td>
          <td
            id="tickerQty0"
          >
            7
          </td>
          <td
            id="tickerDate0"
          >
            Nov 22, 2019
          </td>
        </tr>
        <tr>
          <td
            id="tickerName1"
          >
            MSFT
          </td>
          <td
            id="tickerPrice1"
          >
            135.68
          </td>
          <td>
            Bought
          </td>
          <td
            id="tickerQty1"
          >
            6
          </td>
          <td
            id="tickerDate1"
          >
            Nov 22, 2019
          </td>
        </tr>
      </table>
    </div>
  </div>
</div>
`;
